app/components/subject-picker/template.hbs

```diff
@@ -1,47 +1,34 @@
 <div class="row">
-    {{#each selected as |subject|}}
+    {{#each currentSubjects as |subject index|}}
         <span class="subject">
             {{#each subject as |segment|}}
                 <b>{{segment.text}}</b>
             {{/each}}
-            {{fa-icon 'times' click=(action 'deselect' subject)}}
+            {{fa-icon 'times' click=(action 'deselect' index)}}
         </span>
     {{/each}}
 </div>
 
 <div class="row">
+    {{#each columns as |column tier|}}
+        <div class="col-md-4">
+            <ul style="overflow: scroll">
+                {{input value=column.filterText class="form-control" placeholder="Search"}}
+                {{#each column.subjectsSorted as |subject|}}
+                    <li {{action 'select' subject tier}} class={{if (eq subject column.selection) 'selected'}}>
```

Comment:
Going with [listbox](https://www.w3.org/TR/wai-aria/roles#listbox) & listitem, since they seem like the best fit